src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -1018,7 +1018,27 @@ def getDocsFromMongo(self, mQueryDict, dbColl, limit=1000):
                 self.logger.exception(msg)
                 raise ex
         else:
-            self.logger.info("Query: '%s' did not return any records from MongoDB", mQueryDict)
+            self.logger.info("%s Query: '%s' did not return any records from MongoDB", dbColl.name, mQueryDict)
+
+    def getTransferInfo(self, reqName):
+        """
+        Searches and reads a document from MongoDB in all collections related to
+        the MSOutput service. It is supposed to be called only by MSManager e.g.:
+           transferDoc = self.msOutputProducer.getTransferInfo(reqName)
+        And the output to be served to the REST interface, so that all request
+        transfer records can be tracked.
+        :param reqName: The name of the request to be searched for
+        :return: a list of all msOutDocs with the record if it exists or None otherwise
+        """
+        mQueryDict = {'RequestName': reqName}
+        result = []
+        stripKeys = ['_id', 'producerDoc']
+        for dbColl in [self.msOutRelValColl, self.msOutNonRelValColl]:
```

Comment:
On this method, I have a few ideas that would improve the performance of this API:
* return the document as soon as a document is retrieved (and in that case, we should start with the non-relval collection)
* instead of calling getDocsFromMongo method (which does a conversion to MSOutputTemplate), how about we go straight with `self.msOutNonRelValColl.find(mQueryDict)`?

If performance difference is negligible, then there is no need to change anything.

Reply:
Done